nima/tests/integration/webclient/webclient/src/test/java/io/helidon/nima/webclient/http1/ClientRequestImplTest.java

```diff
@@ -0,0 +1,411 @@
+        List<Http1ClientResponse> responseList = new ArrayList<Http1ClientResponse>();
+        // create connections beyond the queue size limit
+        for (int i = 0; i < connectionQueueSize + 1; ++i) {
+            Http1ClientRequest request = client.method(Http.Method.PUT).path("http://localhost:" + port + "/test");
+            connectionList.add(((ClientRequestImpl) request).getConnection(true));
+            request.connection(connectionList.get(i));
+            responseList.add(request.request());
+        }
+
+        // Queue up all connections except the last one because it exceeded the queue size limit
+        for (Http1ClientResponse response : responseList) {
+            response.close();
+        }
+
+        // dequeue all the created connections
+        ClientConnection connection = null;
+        Http1ClientResponse response = null;
+        for (int i = 0; i < connectionQueueSize + 1; ++i) {
+            Http1ClientRequest request = client.method(Http.Method.PUT).path("http://localhost:" + port + "/test");
+            connection = ((ClientRequestImpl) request).getConnection(true);
+            request.connection(connection);
+            response = request.request();
+            if (i < connectionQueueSize) {
+                // Verify connections that are dequeued
+                assertThat("Failed on connection index " + i, connection, is(connectionList.get(i)));
+            } else {
+                // Verify that the last connection was not dequeued but created as new, because it exceeded the queue size limit
+                assertThat(connection, is(not(connectionList.get(i))));
+            }
+        }
+
+        // The queue is currently empty so check if we can add the last created connection into it.
+        response.close();
+        Http1ClientRequest request = client.method(Http.Method.PUT).path("http://localhost:" + port + "/test");
+        ClientConnection connectionNow = ((ClientRequestImpl) request).getConnection(true);
+        request.connection(connectionNow);
+        Http1ClientResponse responseNow = request.request();
+        // Verify that the connection was dequeued
+        assertThat(connectionNow, is(connection));
+    }
+
+    private static void validateSuccessfulResponse(Http1Client client) {
+        String requestEntity = "Sending Something";
+        Http1ClientRequest request = client.method(Http.Method.PUT).path("http://localhost:" + port + "/test");
+        Http1ClientResponse response = request.submit(requestEntity);
+
+        assertThat(response.status(), is(Http.Status.OK_200));
+        assertThat(response.entity().as(String.class), is(requestEntity));
+    }
+
+    private static void validateFailedResponse(Http1Client client, String errorMessage) {
+        String requestEntity = "Sending Something";
+        Http1ClientRequest request = client.method(Http.Method.PUT).path("http://localhost:" + port + "/test");
+        final IllegalStateException ie =
```

Comment:
Please do not use `final` in local variable declarations

Reply:
updated.